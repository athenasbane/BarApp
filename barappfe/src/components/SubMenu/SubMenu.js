import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Breadcrumbs, Link, makeStyles } from '@material-ui/core';
import { connect } from 'react-redux';

const useStyles = makeStyles({
  root: {
    marginTop: '40px',
  },
});

// const breadcrumbData = [{
//     name: 'Full Menu',
//     path: '#top'
// }, {
//     name: 'Drinks',
//     path: '/drinks'
// }, {
//     name: 'Food',
//     path: '/food'
// }];

const SubMenu = ({ products }) => {
  const breadcrumbData = (products) => {
    const unique = [...new Set(products.map((item) => item.category))];
    if (unique.length > 1) {
      return unique.map((product) => ({
        name: product.charAt(0).toUpperCase() + product.slice(1),
        path: `#${product}`,
      }));
    }
    return products.map((products) => ({
      name: products.subCategory,
      path: `#${products.subCategory}`,
    }));
  };

  const menuData = breadcrumbData(products);

  const links = menuData.map((link) => (
    <Link key={link.name} color="inherit" href={link.path}>
      {link.name}
    </Link>
  ));

  const classes = useStyles();
  return (
    <Grid id="#top" className={classes.root} container item xs={12}>
      <Grid item xs={12}>
        <Breadcrumbs maxItems={3} aria-label="breadcrumb">
          {links}
        </Breadcrumbs>
      </Grid>
    </Grid>
  );
};

SubMenu.propTypes = {
  products: PropTypes.arrayOf(PropTypes.object),
};

const mapStateToProps = (state) => ({
  products: state.products.productData,
});

export default connect(mapStateToProps)(SubMenu);
