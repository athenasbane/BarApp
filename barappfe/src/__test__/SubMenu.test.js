import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import { SubMenu } from '../components/SubMenu/SubMenu';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore();

describe('SubMenu Tests', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <SubMenu />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });
}); 