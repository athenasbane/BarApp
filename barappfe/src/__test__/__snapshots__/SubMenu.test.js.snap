// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubMenu Tests renders correctly 1`] = `ShallowWrapper {}`;
